src/main/java/org/mule/extension/ftp/internal/source/FtpDirectoryListener.java

```diff
@@ -227,8 +235,8 @@ private void refreshMatcher() {
   }
 
   @Override
-  public void onRejectedItem(Result<String, FtpFileAttributes> result, SourceCallbackContext callbackContext) {
-    LOGGER.info("File processing was rejected for path: {}", result.getOutput());
+  public void onRejectedItem(Result<InputStream, FtpFileAttributes> result, SourceCallbackContext callbackContext) {
+    closeQuietly(result.getOutput());
```

Comment:
If an item is rejected by the polling context you should at least inform it. If necessary, also close it.

Reply:
added.